File: software/mobile/android/core-ble/src/main/kotlin/com/avlyubimov/svc/core/ble/BleTransport.kt
package com.avlyubimov.svc.core.ble

import com.avlyubimov.svc.core.model.ConnectionState
import com.avlyubimov.svc.core.model.DiscoveredDevice
import kotlinx.coroutines.flow.StateFlow

interface BleTransport {
    val connectionState: StateFlow<ConnectionState>
    val devices: StateFlow<List<DiscoveredDevice>>
    fun startScan()
    fun stopScan()
    fun connect(deviceId: String)
    fun write(characteristicUuid: String, data: ByteArray)
}

File: software/mobile/ios/SVCMobile/Sources/BLE/BLETransport.swift
import Foundation

protocol BLETransport: AnyObject {
    var onDevicesChanged: (([DiscoveredDevice]) -> Void)? { get set }
    var onConnectionChanged: ((ConnectionState) -> Void)? { get set }
    func startScan()
    func stopScan()
    func connect(to id: UUID)
    func write(_ data: Data, characteristic: UUID)
}
